### PR TITLE
501 Story on Pre-Survey

### LINKED ISSUE #501: Story: Conditional Grouping Based on Pre-Survey
# Template Instructions

Use this template to create a new story item.

1. Copy this template and paste it into a new item project item under backlog.
2. Fill in the placeholders with your own information.
3. Remove these instructions from the item (the horizontal line and above)
4. Add any additional details or context that may be helpful.5. Assign the issue to a team member or yourself.
5. Use the checkboxes to track progress and ensure all requirements are met before marking the issue as complete.
6. Use the checkboxes to track progress and ensure all requirements are met before marking the issue as complete.

---

# <Story Title>

As a **role**, I want **goal**, so that **reason**.

Associated Epic:

## Definition of Done

- [ ] Acceptance criteria defined
- [ ] Solution tasks specified
- [ ] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**
- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**
- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**

## Solution Tasks

- [ ] [Task 1]
- [ ] [Task 2]
- [ ] [Task 3]

### PR DESCRIPTION
**Pull request by apr8760**
# Balance grouping of Imagine MC pre-survey questions/responses.

How to test:
- Create new Postman POST request with JSON body (provided below as requestBody.txt), to endpoint "http://localhost:5005/imagine/presurvey".
- Open DB table imagine23 in DataGrip (table found in folder testing/public/tables).
- Attach debugger for backend + send request.
- Refresh imagine23 table to view newly populated row.
- Update the Postman request body with a new userID (like say test2) and send request again.
- Repeat to populate the table until you can clearly see user responses correctly balance between the 4 sections, "control", "experiential", "discomfortCountNonPOC" and "discomfortCountPOC" (see picture imagine23Populated)

imagine23Populated
[requestBody.txt](https://github.com/all-rit/ALL/files/14438523/requestBody.txt)

Closes #501 

- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

**Comment by heathermoses** [comment]
It looks like there's a lot of reformatted files in this PR that doesn't necessarily need to be there, would it be possible to remove those files from this PR?

File: server/services/ImagineService.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (heathermoses) [inline server/services/ImagineService.js]: Remove eslint

**Review by heathermoses — APPROVED** [review]
lgtm

**Review by dm9965 — APPROVED** [review]
Tested API call through Postman as instructed, POST call properly populates the Imagine 2023 table with preSurvey data. Everything looks good to me!

**Pull request merged**